### ISSUE TITLE
Response that should contain a body errors for 201 and 204 Status Codes

### ISSUE THREAD

**Issue by joshuamil** [42Crunch/vscode-openapi#165]
When defining an empty body response for a 201 (204, etc.) Status Code, which is not intended to return a body, and which OpenAPI supports, the VS Code plugin is returning an error message with a 12 point deduction. It also seems to have issues with maxLength, minLength, and pattern combinations like such:

pattern: /^$/
minLength: 0
maxLength: 0

and requires a "pattern" even when the minLength and maxLength are both set to zero

Basically, there's no way to define an empty body without there being errors reported.

Also note that the "Audit Report" will report errors in example matching that are not shown in the list of problems and that do not appear to be accurate. 

![image](https://user-images.githubusercontent.com/1080695/158828279-600cdec3-de88-4256-8676-89ce4effac7d.png)

**Comment by ak1394** [comment]
Thanks for reporting, I'll raise the issue with the team working on the audit functionality.
